x-rays: harden against incomplete fingerprints

sbelak opened the pull request:
Correctly handles the situation where timestamp fingerprinting somehow failed and we just stored nil (incomplete fingerprints are not ideal, but we should accept the cruel & messy world where this happens).

Fixes #11603

salsakran committed:
bump version!

paulrosenzweig committed:
Preserve template-tags key order (#11582)

paulrosenzweig committed:
Fix SQL editor typing interruption and click positioning (#11596)

paulrosenzweig committed:
Ensure Question.isDirtyComparedTo works for native queries (#11579)

tlrobinson committed:
Remove useless MetabaseApi e2e snapshot test (#11632)

mazameli committed:
fix info icon spacing on dashcards (#11627)

mazameli committed:
fix trend arrow icon spacing (#11629)

mazameli committed:
Document the hide_parameters URL option in the embedding doc (#11622)

* document the hide_parameters url option [ci skip]

* add a missing s

Co-Authored-By: Damon P. Cortesi <d.lifehacker@gmail.com>

* add info to 12-public-links.md

Co-authored-by: Damon P. Cortesi <d.lifehacker@gmail.com>

tlrobinson committed:
update older-than test to not use hard coded date (#11642) (#11644)

* update older-than test to not use hard coded date

* update docstring to clarify that  compares times to now

* final fix for this bug

Co-authored-by: Daniel Higginbotham <daniel@flyingmachinestudios.com>

camsaul committed:
Update Mac App build instructions [ci skip] (#11645)

* Update Mac App build instructions [ci skip]

* [ci skip]

lins05 committed:
Update personal collection name when user name is updated (#11379)

* Update personal collection name when user name is updated

* Removed one line dev code

* Fixed lint errors

* Updated according to CR

* CR

paulrosenzweig committed:
Fix trend UI issues (#11649)

walterl committed:
Port note to IE11 users from EE docs (#11583)

* Port note to IE11 users from EE's docs

* Update docs/enterprise-guide/full-app-embedding.md

Co-Authored-By: Maz Ameli <maz@metabase.com>

Co-authored-by: Maz Ameli <maz@metabase.com>

camsaul committed:
Fix Mac App Title Bar [ci skip] (#11656)

sbelak committed:
x-rays: harden against incomplete fingerprints

paulrosenzweig committed:
strip metadata from segment/metric before passing to segment/metric form

paulrosenzweig committed:
remove "restore()", add manual cleanup

camsaul committed:
Fix Google Analytics break out by Date:Month (#11681)

camsaul committed:
Reconcile temporal types in BiqQuery parameterized SQL field filters (#11678)

walterl committed:
Fix LDAP groups sync regression (#11654)

camsaul committed:
Fix testing connections for new Presto DBs [ci presto] (#11687)

camsaul reviewed (approved):
WFM

Can you rebase this against `release-0.34.x` and merge the PR into that branch so we can release this as part of 0.34.1?

camsaul committed:
core(pulse): fix year format (year-of-era vs week-based-year) (#11692)

Related Documentation of date format:
https://docs.oracle.com/javase/8/docs/api/java/time/format/DateTimeFormatter.html

In pulse notification I found out (works since yesterday until
tonight) that a timetamp formatted by a Pulse would output "Dec 31,
2020" for today's date (instead of "Dec 31, 2019").
This is because of the confusion between "YYYY" (the week based year,
this week is the 1st week of 2020) and "yyyy" formatting (the era
based year - thus the commonly expected year for a single day format).

Co-authored-by: Paul B. <paul+gh@bonaud.fr>

camsaul committed:
Fix SQL Server field filters against boolean Fields (#11691)

* Fix BigQuery driver from Leiningen [ci bigquery] [ci sparksql] (#11650)

* Fix SQL Server field filters against boolean Fields.
[ci drivers]

* Test fixes for drivers with no TIME data type
[ci oracle] [ci presto] [ci redshift] [ci snowflake] [ci sparksql]

* Fix boolean field filters for Oracle [ci oracle] [ci sparksql]

sbelak commented:
Superseded by #11698
